src/openms/include/OpenMS/CHEMISTRY/TheoreticalSpectrumGeneratorXLMS.h

```diff
@@ -0,0 +1,141 @@
+// ETH Zurich, and Freie Universitaet Berlin 2002-2017.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Eugen Netz $
+// $Authors: Eugen Netz $
+// --------------------------------------------------------------------------
+
+
+#ifndef THEORETICALSPECTRUMGENERATORXLMS_H
+#define THEORETICALSPECTRUMGENERATORXLMS_H
+
+#include <OpenMS/CHEMISTRY/Residue.h>
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+#include <OpenMS/KERNEL/StandardTypes.h>
+
+
+namespace OpenMS
+{
+  class AASequence;
+
+  /**
+      @brief Generates theoretical spectra for cross-linked peptides with various options
+
+  @htmlinclude OpenMS_TheoreticalSpectrumGeneratorXLMS.parameters
+
+      @ingroup Chemistry
+  */
+  class OPENMS_DLLAPI TheoreticalSpectrumGeneratorXLMS :
+    public DefaultParamHandler
+  {
+public:
+
+    /** @name Constructors and Destructors
+    */
+    //@{
+    /// default constructor
+    TheoreticalSpectrumGeneratorXLMS();
+
+    /// copy constructor
+    TheoreticalSpectrumGeneratorXLMS(const TheoreticalSpectrumGeneratorXLMS & source);
+
+    /// destructor
+    virtual ~TheoreticalSpectrumGeneratorXLMS();
+    //@}
+
+    /// assignment operator
+    TheoreticalSpectrumGeneratorXLMS & operator=(const TheoreticalSpectrumGeneratorXLMS & tsg);
+
+    /** @name Acessors
+     */
+    //@{
+    /// returns a spectrum with b and y peaks
```

Comment:
please document using `@brief` and `@param` for all input parameters as this is a public function